### PR TITLE
Don't depend on credo and dialyxir in prod

### PR DESCRIPTION
**Pull request by axelson**
Any dependencies harald has that are in the prod environment become transitive dependencies of any applications that depend on Harald which means that if the app already depends on a version of dialyxir then a conflict will occur.

Change credo and dialyxir to run in the TEST env on CI

### CODE REVIEW

**Comment by danielspofford** [comment]
@axelson CI failed for this PR because it was from a fork and CircleCI withheld Coveralls secret from the CI run. I have since merged a change into master that only does coveralls on master. The hope is that a PR like this can still pass CI tests, no need for coveralls. Then if the PR gets merged, that the internal CI run on master can run Coveralls because CircleCI won't withhold the creds. That is the hope anyway.

Can you force push a rebase off master with those changes so we can see if you get a green CI light?

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
       - run: mix deps.get
       - run: mix compile --warnings-as-errors
       - run: MIX_ENV=test mix test
-      - run: mix credo
+      - run: MIX_ENV=test mix credo
       - run: mix format --check-formatted
       - run: mix dialyzer --plt
```
Comment (danielspofford) [inline .circleci/config.yml]: This will need to be prefixed with `MIX_ENV=test` as well.

**Comment by axelson** [comment]
Hmm, looks like we're running into https://github.com/whatyouhide/stream_data/issues/130 which doesn't appear to have a resolution

**Comment by danielspofford** [comment]
I think this can be worked around by Harald - I'll give it some thought.

**Review by danielspofford — APPROVED** [review]

**Pull request merged**